File: data/code_pairs/cpp/1434.cpp
#include <iostream>
#include <vector>
#include <cmath>

void gauleg(const double x1, const double x2, std::vector<double>& x, std::vector<double>& w, const int n) {
    const double EPS = 3.0e-14;
    const int m = (n + 1) / 2;
    const double xm = 0.5 * (x2 + x1), xl = 0.5 * (x2 - x1);

    for (int i = 0; i < m; ++i) {
        double z = cos(M_PI * (i + 0.75) / (n + 0.5));
        double p1 = 1.0, p2 = 0.0, pp, z1;

        do {
            double p3 = p2;
            p2 = p1;
            p1 = ((2.0 * (i + 1) - 1.0) * z * p2 - i * p3) / (i + 1);

            for (int j = 1; j < n; ++j) {
                p3 = p2;
                p2 = p1;
                p1 = ((2.0 * (j + 1) - 1.0) * z * p2 - j * p3) / (j + 1);
            }

            pp = n * (z * p1 - p2) / (z * z - 1.0);
            z1 = z;
            z = z1 - p1 / pp;
        } while (fabs(z - z1) > EPS);

        x[i] = xm - xl * z;
        x[n - i - 1] = xm + xl * z;
        w[i] = 2.0 * xl / ((1.0 - z * z) * pp * pp);
        w[n - i - 1] = w[i];
    }
}

int main() {
    int N = 5;
    double X1 = 0.0, X2 = 1.0;
    std::vector<double> X(N), W(N);

    gauleg(X1, X2, X, W, N);

    std::cout << "X: ";
    for (const auto& xi : X) std::cout << xi << " ";
    std::cout << std::endl;

    std::cout << "W: ";
    for (const auto& wi : W) std::cout << wi << " ";
    std::cout << std::endl;

    return 0;
}

File: data/code_pairs/fortran/1434.f90
REAL*8 FUNCTION func(x)
      REAL*8, INTENT(IN) :: x
      func = x * x
      END FUNCTION func

      SUBROUTINE GAULEG(X1,X2,X,W,N)
      IMPLICIT NONE
      REAL*8, INTENT(IN) :: X1, X2
      REAL*8, INTENT(OUT) :: X(N), W(N)
      REAL*8 :: EPS, XM, XL, Z, P1, P2, P3, PP, Z1
      INTEGER :: M, I, J
      PARAMETER (EPS=3.D-14)
      M=(N+1)/2
      XM=0.5D0*(X2+X1)
      XL=0.5D0*(X2-X1)
      DO I=1,M
          Z=COS(3.141592654D0*(I-.25D0)/(N+.5D0))
1         CONTINUE
              P1=1.D0
              P2=0.D0
              DO J=1,N
                  P3=P2
                  P2=P1
                  P1=((2.D0*J-1.D0)*Z*P2-(J-1.D0)*P3)/J
              END DO
              PP=N*(Z*P1-P2)/(Z*Z-1.D0)
              Z1=Z
              Z=Z1-P1/PP
              IF(ABS(Z-Z1).GT.EPS) GO TO 1
              X(I)=XM-XL*Z
              X(N+1-I)=XM+XL*Z
              W(I)=2.D0*XL/((1.D0-Z*Z)*PP*PP)
              W(N+1-I)=W(I)
      END DO
      END SUBROUTINE GAULEG

      PROGRAM TestGauleg
      IMPLICIT NONE
      REAL*8, ALLOCATABLE :: X(:), W(:)
      INTEGER :: N
      REAL*8 :: X1, X2
      
      N = 5
      X1 = 0.0
      X2 = 1.0
      ALLOCATE(X(N), W(N))
      
      CALL GAULEG(X1, X2, X, W, N)
      
      PRINT *, 'X:', X
      PRINT *, 'W:', W
      
      DEALLOCATE(X, W)
      END PROGRAM TestGauleg